crates/ursa-pod/benches/e2e.rs

```diff
@@ -267,5 +292,282 @@ mod http_hyper {
+        type Error = rustls::Error;
+
+        fn start_server(self) -> Result<Self::Server, Self::Error> {
+            Ok(server_config().into())
+        }
+
+        fn start_client(self) -> Result<Self::Client, Self::Error> {
+            Ok(client_config().into())
+        }
+    }
+
+    pub async fn server_loop(
+        addr: String,
+        content: &'static [u8],
+        tx_started: tokio::sync::oneshot::Sender<u16>,
+    ) {
+        let mut server = Server::builder()
+            .with_tls(TlsProvider)
+            .unwrap()
+            .with_io(addr.as_str())
+            .unwrap()
+            .start()
+            .unwrap();
+
+        tx_started
+            .send(server.local_addr().unwrap().port())
+            .unwrap();
+
+        while let Some(mut conn) = server.accept().await {
+            let content_clone = content.clone();
+            task::spawn(async move {
+                loop {
+                    match conn.accept_bidirectional_stream().await {
+                        Ok(Some(stream)) => {
+                            task::spawn(async {
+                                let handler = UfdpHandler::new(
+                                    stream,
+                                    DummyBackend {
+                                        content: content_clone,
+                                    },
+                                    0,
+                                );
+                                if let Err(e) = handler.serve().await {
+                                    println!("server error: {e:?}");
+                                }
+                            });
+                        }
+                        Ok(None) => break,
+                        Err(s2n_quic::connection::Error::Closed { .. }) => break,
+                        Err(e) => panic!("{e:?}"),
+                    }
+                }
+            });
+        }
+    }
+
+    pub async fn client_loop(addr: String, iterations: usize) {
+        let mut tasks = vec![];
+        let client = Client::builder()
+            .with_tls(TlsProvider)
+            .unwrap()
+            .with_io("0.0.0.0:0")
+            .unwrap()
+            .start()
+            .unwrap();
+        let addr: SocketAddr = addr.parse().unwrap();
+        let mut connection = client
+            .connect(Connect::new(addr).with_server_name("localhost"))
+            .await
+            .unwrap();
+        for _ in 0..iterations {
+            let stream = connection.open_bidirectional_stream().await.unwrap();
+            let task = task::spawn(async move {
+                let mut client = UfdpClient::new(stream, CLIENT_PUB_KEY, None).await.unwrap();
+                client.request(CID).await.unwrap();
+            });
+            tasks.push(task);
+        }
+        join_all(tasks).await;
+    }
+}
+
+#[cfg(feature = "bench-quic")]
+mod tls_utils {
+    use std::sync::Arc;
+
+    pub struct SkipServerVerification;
```

Comment:
I forgot, I'm going to remove this because it would be better to not skip the certificate verification so we can get a better idea of the performance.

Reply:
Since the iterations in the bench are multiplexed over a single quic connection, this would only affect the initial connection right?

Reply:
Yes, that's right.